### PR TITLE
Fix issue 15033 -  Element type of float iota is double

### PR DESCRIPTION
**Pull request by nordlow**
https://issues.dlang.org/show_bug.cgi?id=15033

Use CommonType in iota()

See also: http://forum.dlang.org/post/ahjmgdslmiakoxszynty@forum.dlang.org

### CODE REVIEW

File: std/range/package.d [hunk 1]
```diff
@@ -4625,9 +4625,9 @@ body
 
 /// Ditto
 auto iota(B, E)(B begin, E end)
-if (isFloatingPoint!(CommonType!(B, E)))
+    if (isNumeric!(CommonType!(B, E)))
```
Comment (trikko) [inline std/range/package.d]: I guess you should leave isFloatingPoint. Some lines after there's a version for integer types.

Comment (nordlow) [inline std/range/package.d]: Done.

**Comment by MetaLang** [comment]
LGTM. FreeBSD seems to be randomly failing a lot lately for some weird reason... Looks unrelated to this pull.

**Comment by nordlow** [comment]
I guess we should hava a unittest for this that failed before the change, right?

**Comment by schveiguy** [comment]
The test should be what is in the bug report:
```
    static assert(is (ElementType!(typeof(iota(0f))) == float));
```

**Comment by schveiguy** [comment]
Updated your title/summary to reflect bug report. You should also set up your log message to say "Fix 15033" and bugzilla will automatically close the bug for you.

**Comment by quickfur** [comment]
LGTM

**Comment by nordlow** [comment]
Done.

**Comment by burner** [comment]
LGTM

**Comment by burner** [comment]
Auto-merge toggled on

**Pull request merged**